docs/02-app/01-building-your-application/01-routing/11-middleware.mdx

```diff
@@ -121,11 +123,24 @@ The `NextResponse` API allows you to:
 - Set response cookies
 - Set response headers
 
+<AppOnly>
+
 To produce a response from Middleware, you can:
 
-1. `rewrite` to a route ([Page](/docs/pages/building-your-application/routing/pages-and-layouts) or [Edge API Route](/docs/pages/building-your-application/routing/api-routes)) that produces a response
+1. `rewrite` to a route ([Page](/docs/app/building-your-application/routing/pages-and-layouts) or [Route Handler](/docs/app/building-your-application/routing/api-routes)) that produces a response
```

Comment:
This link is 404

https://nextjs.org/docs/app/building-your-application/routing/api-routes

Reply:
Just fixed it. I accidentally switched the api routes and router handlers links